src/main/java/bio/terra/controller/DataRepositoryServiceApiController.java

```
@@ -0,0 +1,134 @@
+            StudyService studyService,
+            DatasetRequestValidator datasetRequestValidator,
+            DatasetService datasetService,
+            SamClientService samService,
+            IngestRequestValidator ingestRequestValidator,
+            ApplicationConfiguration appConfig,
+            FileService fileService,
+            DrsService drsService
+    ) {
+        this.objectMapper = objectMapper;
+        this.request = request;
+        this.jobService = jobService;
+        this.studyRequestValidator = studyRequestValidator;
+        this.studyService = studyService;
+        this.datasetRequestValidator = datasetRequestValidator;
+        this.datasetService = datasetService;
+        this.samService = samService;
+        this.ingestRequestValidator = ingestRequestValidator;
+        this.appConfig = appConfig;
+        this.fileService = fileService;
+        this.drsService = drsService;
+    }
+
+/*
+    @InitBinder
+    protected void initBinder(final WebDataBinder binder) {
+        binder.addValidators();
+    }
+*/
+
+    @Override
+    public Optional<ObjectMapper> getObjectMapper() {
+        return Optional.ofNullable(objectMapper);
+    }
+
+    @Override
+    public Optional<HttpServletRequest> getRequest() {
+        return Optional.ofNullable(request);
+    }
+
+    private AuthenticatedUserRequest getAuthenticatedInfo() {
+        return AuthenticatedUserRequest.from(getRequest(), appConfig.getUserEmail());
+    }
+
+    @Override
+    public ResponseEntity<DRSAccessURL> getAccessURL(@PathVariable("object_id") String objectId,
+                                                     @PathVariable("access_id") String accessId) {
+        // TODO: this has to return DRSError - not an ErrorModel
+        // Have to figure out how to do that
+
+        // We never give out access ids, so by definition, the input is invalid.
+        throw new ValidationException("Invalid access_id: '" + accessId + "'");
```

or should we throw a NOT_IMPLEMENTED exception?

---

I don't think so. DRS has this optional, two-step flow where the client does a lookup with the `GET /object/{objectid}`. That GET might return a `access_id`. The client can then do a `GET /object/{objectid}/access/{access_id}` to get an URL to read the bytes. The goal of the two-step flow is to allow the get object without generating signed URLs for all of the locations of the object.

Since we do not use signed URLs, we never generate an access_id. So I think it is always an error for the client to call this endpoint.